src/otx/v2/adapters/torch/mmengine/mmdet/modules/evaluation/evaluator.py

```diff
@@ -0,0 +1,469 @@
+        width (int): image width
+        padding (int, optional): padding added to each side of the bounding box. Defaults to 1.
+
+    Returns:
+        np.ndarray: sanitized bounding boxes with shape (4, )
+    """
+    x1, y1, x2, y2 = bbox.astype(int)
+    x1 = max(0, x1 - padding)
+    y1 = max(0, y1 - padding)
+    x2 = min(width, x2 + padding)
+    y2 = min(height, y2 + padding)
+    return np.array([x1, y1, x2, y2])
+
+
+def mask_iou(det: tuple[np.ndarray, BitmapMasks], gt_masks: PolygonMasks, iou_thr: float) -> np.ndarray:
+    """Compute the intersection over union between the detected masks and ground truth masks.
+
+    Args:
+        det (Tuple[np.ndarray, BitmapMasks]): detected bboxes and masks
+        gt_masks (PolygonMasks): ground truth masks
+        iou_thr (float): IoU threshold
+
+    Note:
+        It first compute IoU between bounding boxes, then compute IoU between masks
+        if IoU between bounding boxes is greater than 0.
+        Detection mask is resized to detected bounding box size and
+        padded to the same size as ground truth mask in order to compute IoU.
+
+    Returns:
+        np.ndarray: iou between detected masks and ground truth masks
+
+    """
+    det_bboxes, det_masks = det
+    gt_bboxes = gt_masks.get_bboxes("hbox")
+    img_h, img_w = gt_masks.height, gt_masks.width
+    ious = bbox_overlaps(torch.from_numpy(det_bboxes), gt_bboxes.tensor, mode="iou").numpy()
+    ious[ious < iou_thr] = 0.0
+    if not ious.any():
+        return ious
+    # NOTE: further speed optimization (vectorization) could be done here
+    for coord in np.argwhere(ious):
+        m, n = coord
+        det_bbox, det_mask = sanitize_coordinates(det_bboxes[m], img_h, img_w), det_masks[m]
+        gt_bbox, gt_mask = sanitize_coordinates(gt_bboxes.numpy()[n], img_h, img_w), gt_masks[n]
+        # add padding to det_mask and gt_mask so that they have the same size
+        min_x1 = min(det_bbox[0], gt_bbox[0])
+        min_y1 = min(det_bbox[1], gt_bbox[1])
+        max_x2 = max(det_bbox[2], gt_bbox[2])
+        max_y2 = max(det_bbox[3], gt_bbox[3])
+
+        det_bbox_h, det_bbox_w = det_bbox[3] - det_bbox[1], det_bbox[2] - det_bbox[0]
+        det_mask = det_mask.resize((det_bbox_h, det_bbox_w))
+        det_mask = det_mask.expand(max_y2 - min_y1, max_x2 - min_x1, det_bbox[1] - min_y1, det_bbox[0] - min_x1)
+        gt_mask = gt_mask.crop(gt_bbox)
+        gt_mask = gt_mask.to_bitmap()
+        gt_mask = gt_mask.expand(max_y2 - min_y1, max_x2 - min_x1, gt_bbox[1] - min_y1, gt_bbox[0] - min_x1)
+        # compute iou between det_mask and gt_mask
+        det_mask = det_mask.to_ndarray()
+        gt_mask = gt_mask.to_ndarray()
+        if det_mask.shape != gt_mask.shape:
+            raise ValueDifferenceError(det_mask.shape, gt_mask.shape)
+        ious[m, n] = np.sum(det_mask & gt_mask) / np.sum(det_mask | gt_mask)
+    return ious
+
+
+def tpfpmiou_func(  # pylint: disable=too-many-locals
```

Comment:
again, we don't need pylint anymore.
```suggestion
def tpfpmiou_func(
```

Reply:
Also the function name did not make any sense to me when I first read it. I had to navigate to the docstring to read the details. Would it be possible to make the function name more explicit?

Reply:
I changed this to compare_gts_and_annotations. Any suggestion?